src/Notes/NoteTraits.php

```diff
@@ -143,6 +143,26 @@ public static function possibly_delete_note() {
 		}
 	}
 
+
+	/**
+	 * Update the note if it passes predefined conditions.
+	 *
+	 * @throws NotesUnavailableException Throws exception when notes are unavailable.
+	 */
+	public static function possibly_update_note() {
+		$note = Notes::get_note_by_name( self::NOTE_NAME );
+
+		if ( ! $note ) {
+			return;
+		}
+
+		// Update note content if it's changed.
+		$latest_note_content = self::get_note()->get_content();
+		if ( $note->get_content() !== $latest_note_content ) {
+			$note->set_content( $latest_note_content );
+			$note->save();
+		}
```

Comment:
> I just thought that we should also update other note attributes along the way. This should save us from having to handle it in the future. Remote inbox already works like this.

> I'm happy to have it as a follow-up as well, what do you think

Absolutely, It sounds like a nice enhancement we should have! 👍  It also reminds me of this closed issue:
- https://github.com/woocommerce/woocommerce-admin/issues/6063

I agree we can have another issue to follow up on since this PR is created for #8008.

Reply:
Great! I have filed a follow-up issue https://github.com/woocommerce/woocommerce-admin/issues/8427